src/components/ListItem.jsx

```diff
@@ -1,5 +1,41 @@
+import React, { useState, useEffect } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listPath }) {
+	const { id, totalPurchases, name, dateLastPurchased } = item;
+	const lessThan24HoursSincePurchase =
+		dateLastPurchased &&
+		(new Date() - dateLastPurchased.toDate()) / 3600000 < 24;
+
+	const [isChecked, setIsChecked] = useState(
+		lessThan24HoursSincePurchase || false,
```

Comment:
note: there is weird behavior for useState when you set default values for it. It will not affect `useState` after the first render. There is more to read on this if you want more info: https://react.dev/reference/react/useState#parameters

There isn't anything wrong with the code above but good to know the behavior.

Reply:
Thanks Britt, I originally did not include a default value but got the error `A component is changing an uncontrolled input to be controlled`, when trying to mark something as done. Think it may be because the initial state is undefined?

Reply:
here’s a blog post outlining what you ran into and a couple of options to solve it - https://bobbyhadz.com/blog/react-component-changing-uncontrolled-input

Reply:
Thanks Aditi, I updated the `checked` value to have a fallback value of `false` instead of having it in the state and it seemed to have worked